[android][task-manager] Task manager for bare in Expo

mczernek opened the pull request:
# Why

In order to deprecate ExpoKit, part of #3453 

# Test Plan

We are working on preparing standalone ncl for that. Currently, bare-expo in apps is the best starting point for testing this feature.

mczernek committed:
[WIP]

mczernek committed:
[WIP]

mczernek committed:
[WIP]

mczernek committed:
[WIP].

mczernek committed:
[WIP] yet another.

mczernek committed:
Kind of refactored.

mczernek committed:
Refactoring.

mczernek committed:
Null TaskManager in TaskService only when react instance dies.

mczernek committed:
Revert bare-expo app.

mczernek committed:
Refactoring still.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare

mczernek committed:
Revert bare-expo changes.

mczernek committed:
Remove needlessly created files in bare-expo's ios dir.

mczernek committed:
Change logic determining whether app runs in headless mode.

esamelson commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/apploader/RNHeadlessAppStarter.kt:
```diff
@@ -0,0 +1,55 @@
+      throw IllegalArgumentException("Params must be set with appId!")
+    }
+
+    val reactInstanceManager = (context.applicationContext as ReactApplication).reactNativeHost.reactInstanceManager
```
Are we pretty much guaranteed than in any case (even anyone's customized bare RN app) the context we receive can be cast to `ReactApplication`? I tried quickly to trace back where this context comes from but it's not obvious. 🤔  If not, it might be nice to throw a helpful error here and also be sure to document this assumption.

esamelson commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerInternalModule.java:
```diff
@@ -112,12 +113,12 @@ public String getAppId() {
-  public boolean isRunningInHeadlessMode() {
-    if (mConstants != null) {
-      return (boolean) mConstants.getConstants().get("isHeadless");
+  private Boolean getBooleanConstant(String key) {
```
maybe unused(?)

esamelson reviewed (commented):
a few small comments on first look, but will probably need to take another pass. Exciting stuff 😄 Curious to see what the installation/setup steps for bare apps will look like, as well, though I'm sure you'll get to that in due course!

One other small note -- for future PR reviews it would be really helpful to have code styling/reformatting in a separate commit as having it all together makes it somewhat more difficult to see what's actually changed. 😅

mczernek committed:
Android and iOS bare app for testing TaskManager.

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskService.java:
```diff
@@ -399,23 +409,31 @@ public void executeTask(TaskInterface task, Bundle data, Error error, TaskExecut
-        appEvents.remove(eventId);
-        sEventsQueues.remove(appId);
+      });
+    } catch (IllegalArgumentException ignored) {
```
`startApp` can also throw `IllegalStateException`

mczernek replied:
Changed to newly created AppConfigurationError. This is checked exception, which better suits our needs.

tsapeta commented on packages/expo-task-manager/android/build.gradle:
```diff
@@ -73,5 +80,12 @@ dependencies {
+  implementation "com.facebook.react:react-native:+"
+
   api "androidx.core:core:1.0.0"
+    implementation "androidx.core:core-ktx:+"
```
different indentation than the lines above? 🤔

tsapeta commented on packages/expo-task-manager/android/build.gradle:
```diff
@@ -73,5 +80,12 @@ dependencies {
   unimodule "unimodules-constants-interface"
   unimodule "expo-app-loader-provider"
 
+  implementation "com.facebook.react:react-native:+"
```
I'm wondering whether we should move our code that depends on React Native to separate package or not. The idea of unimodules architecture was to provide modules that don't depend on the framework that other users may use. For example, adding support for Flutter would require some more work as we don't want to require Flutter's users to depend on RN. What about moving app loading implementation (depending on RN) to the adapter?

tsapeta commented on packages/expo-task-manager/android/src/main/AndroidManifest.xml:
```diff
@@ -1,9 +1,10 @@
         <receiver
             android:name=".TaskBroadcastReceiver"
-            android:exported="false">
+            android:exported="true">
```
Is this intended?

mczernek replied:
Yes and no. It was intended for testing, but should (and will) be reverted.

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerInternalModule.java:
```diff
@@ -48,7 +48,7 @@ public void onCreate(ModuleRegistry moduleRegistry) {
     mTaskService = moduleRegistry.getSingletonModule("TaskService", TaskServiceInterface.class);
 
-    // Register in TaskService.
+    // Register in TaskService
```
Why such change? 😅

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerInternalModule.java:
```diff
@@ -57,6 +57,7 @@ public void onCreate(ModuleRegistry moduleRegistry) {
   @Override
   public void onDestroy() {
     mUIManager.unregisterLifecycleEventListener(this);
+    mTaskService.setTaskManager(null, getAppId(), getAppUrl());
```
Maybe we can use this new static method `removeTaskManager`? It would be more clear what we're doing here.

mczernek replied:
I've added such method to interface and renamed static one. I guess, whenever we can, we should not use explicit static method. Here, we can use here a method which is publicly available via interface, without assuming which implementation we handle. 

We should rather think of a way not to call this static method anywhere else. I'll think about it.

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerPackage.java:
```diff
@@ -10,19 +10,27 @@
 public class TaskManagerPackage extends BasePackage {
+
+  static {
+    AppLoaderProvider.registerLoader("react-native-experience", RNHeadlessAppStarter.class);
```
Again, it's a bit confusing that we register a loader (`registerLoader`) but pass in the class that is called as a starter 😅

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/apploader/RNHeadlessAppStarter.kt:
```diff
@@ -0,0 +1,55 @@
+    appRecords.contains(appId) && appRecords[appId]!!.hasStartedCreatingInitialContext()
+
+  //endregion AppLoader
+}
```
```suggestion
}

```

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/apploader/RNHeadlessAppStarter.kt:
```diff
@@ -0,0 +1,55 @@
+  override fun invalidateApp(appId: String?): Boolean {
+    return if (appRecords.containsKey(appId)) {
+      android.os.Handler(ignore.mainLooper).post {
+        appRecords[appId]!!.destroy()
```
Imho it's not a good practice to use `!!` anywhere - especially when the code execution is postponed by a handler

tsapeta commented on packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/apploader/RNHeadlessAppStarter.kt:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  override fun isRunning(appId: String?): Boolean =
+    appRecords.contains(appId) && appRecords[appId]!!.hasStartedCreatingInitialContext()
```
Same here, I'd replace `!!` with something else

mczernek replied:
In here it is perfectly valid. `contains` returns true only if there is an element in map. Element cannot be null, so it is perfectly safe to use `appRecords[appId]` in here. Even compiler does not produce warnings here.

tsapeta commented on packages/unimodules-task-manager-interface/android/src/main/java/org/unimodules/interfaces/taskManager/TaskManagerInterface.java:
```diff
@@ -5,6 +5,10 @@
 public interface TaskManagerInterface {
+
+  String EVENT_NAME = "TaskManager.executeTask";
+  String E_TASK_SERVICE_NOT_FOUND = "E_TASK_SERVICE_NOT_FOUND";
```
Maybe it's a good occasion to change our error tags and prefix them with `ERR` instead of `E`? 🤔

tsapeta commented on packages/unimodules-task-manager-interface/android/src/main/java/org/unimodules/interfaces/taskManager/TaskServiceInterface.java:
```diff
@@ -82,4 +82,10 @@
   void executeTask(TaskInterface task, Bundle data, Error error, TaskExecutionCallback callback);
+
+  /**
+   *  Checks whether appId is currently being run in headless mode.
```
```suggestion
   *  Checks whether the app with given appId is currently being run in headless mode.
```

tsapeta reviewed (changes requested):
Left some comments. I also think that having a lot of `[WIP]` commits isn't helpful and just look bad even though they will be squashed once merged 😅 So please, keep commits history better next time (you can use WIP commits while they are in progress, but then you should squash them or amend).
Except comments below, could you please take a look why `client_android` job on the CI is failing?

mczernek committed:
Support TaskManager in bare on iOS.

mczernek committed:
Revert "Android and iOS bare app for testing TaskManager."

This reverts commit 3a571f6e42100358502d6fa6446daf8fc30b8837.

mczernek committed:
Fix android client build.

mczernek commented on android/expoview/src/main/java/host/exp/exponent/taskManager/AppRecordInterface.java:
```diff
@@ -1,4 +1,4 @@
-package expo.loaders.provider.interfaces;
+package host.exp.exponent.taskManager;
```
Not needed, but logical. Since I've changed AppLoader to AppStarter, and there is no notion of AppRecordInterface in it, this interface would just stay here for no purpose.

Now, AppRecordInterface is used only internally, in expo client and managed workflow. There is no need to expose this interface.

mczernek commented on android/.idea/jarRepositories.xml:
```diff
@@ -0,0 +1,100 @@
+      <option name="url" value="https://maven.fabric.io/public" />
+    </remote-repository>
+  </component>
+</project>
```
Not sure. My Android Stuido kept creating it, though, so I finally had to decide whether to .gitignore it, or commit. Decided to commit for now, but if that causes any trouble, we should rethink it.

mczernek commented on android/versioned-abis/expoview-abi33_0_0/src/main/java/abi33_0_0/expo/modules/taskManager/TaskService.java:
```diff
@@ -14,6 +14,13 @@
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
+import org.unimodules.core.interfaces.SingletonModule;
```
And that's mostly the case in here.

mczernek commented on android/expoview/src/main/java/host/exp/exponent/taskManager/TaskManagerInternalAppLoader.java:
```diff
@@ -0,0 +1,81 @@
+
+    if (appLoader == null) {
+      throw new IllegalStateException("Cannot execute background task because application loader can't be found.");
+    } else if (params.getAppUrl() == null) {
```
Why so? Personally, I like `if (...) else if` construct, when conditions are actually mutually exclusive. It is clear then, that only one branch will be executed (as intended) w/o looking at code inside each branch.

mczernek committed:
Naming, indentation, formatting.

mczernek committed:
Rename AppStarter back to AppLoader.

mczernek commented on android/expoview/src/main/java/host/exp/exponent/taskManager/TaskManagerInternalAppLoader.java:
```diff
@@ -0,0 +1,81 @@
+import expo.loaders.provider.AppLoaderProvider;
+import host.exp.exponent.headless.HeadlessAppLoader;
+
+public class TaskManagerInternalAppLoader implements HeadlessAppStarter {
```
You are right. I'll get back to AppLoader.

mczernek committed:
Move RNHeadlessAppLoader to react-native adapter.

mczernek committed:
Revert "Revert "Android and iOS bare app for testing TaskManager.""

This reverts commit dc86ea50d1b23f38b898ee8d5810e1bf38a558ce.

mczernek committed:
Prevent from crash in expo-sensors when quitting react activity multiple times.

mczernek committed:
Rename Starter back to loader. Persist loader class for when no React instance is run.

mczernek committed:
Revert immediate background fetch service startup.

mczernek committed:
Backport getting AppLoader in TaskService to versioned code.

mczernek committed:
Correct reactInstanceManager building for headless apps.

mczernek committed:
Revert "Revert "Revert "Android and iOS bare app for testing TaskManager."""

This reverts commit 84c62aa28548b74f4851283a5262f47779d55b91.

mczernek committed:
Update Changelog.

mczernek committed:
Remove test bare app.

This reverts commit b0697dc3935e66a442761ad116dd8d5349226358.

mczernek committed:
Align bare-expo with master.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare

brentvatne committed:
[home] Allow users to paste https experience urls

tsapeta commented on packages/expo-task-manager/ios/EXTaskManager/EXTaskService.m:
```diff
@@ -76,19 +76,19 @@ - (void)registerTaskWithName:(NSString *)taskName
 {
   Class unversionedConsumerClass = [self _unversionedClassFromClass:consumerClass];
-
+  
```
Looks like you have weird formatting settings in Xcode 😅 We'd like to trim whitespaces.

tsapeta commented on packages/expo-task-manager/ios/EXTaskManager/EXTaskService.m:
```diff
@@ -179,16 +179,16 @@ - (NSArray *)getRegisteredTasksForAppId:(NSString *)appId
-                           @"taskType": task.consumer.taskType,
-                           @"options": task.options,
-                           }];
+        @"taskName": taskName,
```
I think the indentation was correct here too

tsapeta commented on packages/expo-task-manager/ios/EXTaskManager/EXTaskService.m:
```diff
@@ -404,10 +404,13 @@ - (void)_addTaskToConfig:(nonnull id<UMTaskInterface>)task
   [tasks setObject:taskDict forKey:task.name];
   [appDict setObject:tasks forKey:@"tasks"];
-  [appDict setObject:task.appUrl forKey:@"appUrl"];
+  if(task.appUrl)
```
```suggestion
  if (task.appUrl) {
```

tsapeta commented on packages/expo-task-manager/ios/EXTaskManager/EXTaskService.m:
```diff
@@ -404,10 +404,13 @@ - (void)_addTaskToConfig:(nonnull id<UMTaskInterface>)task
   [appDict setObject:tasks forKey:@"tasks"];
-  [appDict setObject:task.appUrl forKey:@"appUrl"];
+  if(task.appUrl)
+  {
```
```suggestion
```

tsapeta commented on packages/unimodules-task-manager-interface/android/src/main/java/org/unimodules/interfaces/taskManager/TaskManagerInterface.java:
```diff
@@ -5,6 +5,10 @@
 public interface TaskManagerInterface {
+
+  String EVENT_NAME = "TaskManager.executeTask";
+  String E_TASK_SERVICE_NOT_FOUND = "ERR_TASK_SERVICE_NOT_FOUND";
```
```suggestion
  String ERR_TASK_SERVICE_NOT_FOUND = "ERR_TASK_SERVICE_NOT_FOUND";
```

I think we can change that name so it matches its value.

tsapeta commented on yarn.lock:
```diff
@@ -7036,6 +7036,24 @@ expo-cli@^3.11.1:
     "@expo/traveling-fastlane-darwin" "1.11.0"
     "@expo/traveling-fastlane-linux" "1.11.0"
 
+expo-module-scripts@~1.2.0:
```
I'm not sure but this change in `yarn.lock` shouldn't be there. If I recall correctly, this means something installed `expo-module-scripts` from outside of the workspace. Could you check whether the version of `expo-module-scripts` in the workspace matches the semver pattern that we use in the packages you changed?

mczernek replied:
You might be right. It disappeared after yet another `yarn`, so probably there was some misalignment in versions.

tsapeta commented on packages/expo-task-manager/package.json:
```diff
@@ -34,10 +34,11 @@
     "@unimodules/core": "*",
-    "unimodules-task-manager-interface": "*"
+    "unimodules-task-manager-interface": "*",
+    "unimodule-app-provider-interface": "*"
```
```suggestion
```
We don't have such module 😅

tsapeta reviewed (changes requested):
Another series of comments 😄

mczernek commented on android/expoview/src/main/java/host/exp/exponent/headless/InternalHeadlessAppLoader.java:
```diff
@@ -286,10 +279,9 @@ private RNObject startReactInstance(final Exponent.StartReactInstanceDelegate de
-        IS_HEADLESS_KEY, true
+      MANIFEST_URL_KEY, mManifestUrl,
+      LINKING_URI_KEY, linkingUri,
+      INTENT_URI_KEY, mIntentUri
```
As far as I can tell, expo-constants has no notion of this value. There is no key for it in expo-constants, nor any notion of it in documentation.

mczernek committed:
Minor fixes: formatting, typos, naming.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare

cruzach committed:
[github] fix stale bot part 2

mmeyer724 committed:
[templates] Add .expo/ to gitignore for bare templates (#6811)

brentvatne committed:
[docs] Add missing end quote to config doc

EvanBacon committed:
Create styled-components guide (#6961)

* Create using-styled-components.md

* Update docs/pages/versions/unversioned/guides/using-styled-components.md

Co-Authored-By: Juwan Wheatley <datwheat@gmail.com>

Co-authored-by: Juwan Wheatley <datwheat@gmail.com>

SaeedZhiany committed:
[android][local-authentication] Load Android Gradle Plugin conditionally (#6977)

# Why

This wraps the Android Gradle plugin dependency in the buildscripts section of android/build.gradle in a conditional:

```
if (project == rootProject) {
    // ... (dependency here)
}
```

The Android Gradle plugin is only required when opening the project stand-alone, not when it is included as a dependency. By doing this, the project opens correctly in Android Studio, and it can also be consumed as a native module dependency from an application project without affecting the app project (avoiding unnecessary downloads/conflicts/etc).

for more info, you can refer to [this thread](https://github.com/facebook/react-native/pull/25569) and especially [this comment.](https://github.com/facebook/react-native/pull/25569#issuecomment-532504277)

# How

I used a condition to check this module is the root project opened in IDE or it's just a module used by another root project; then based on the result of the condition I loaded the AGP for building the module. So the AGP will be downloaded and used only when the module is opened as a standalone project (for developing the module itself)

# Test Plan

I'm currently using AGP 3.5.3 in one of my projects. before this change when I try to build my project, Android Gradle Plugin 3.5.1 is also downloaded to build this module which is completely unnecessary.
After this change, the AGP 3.5.1 is not downloaded and this module still built successfully. please be aware this change has no break changes for users and developers of this module and it's a completely safe change. you can check more successful similar PRs on famous RN libraries from the thread I linked above.

beaur committed:
[local-authentication] Add more options to the iOS authenticateAsync options (#6891)

# Why

This resolves an open issue [#4799](https://github.com/expo/expo/issues/4799).  I also added the option to customize the cancel label.

# How

I reviewed the [Apple documentation](https://developer.apple.com/documentation/localauthentication/logging_a_user_into_your_app_with_face_id_or_touch_id?language=objc) and added the appropriate options to select the correct policy to evaluate against.

# Test Plan

I will test this on a number of devices by copying over the source into my node_modules, building the app, and testing on multiple devices.

tsapeta committed:
[local-authentication] Rebuild local authentication package

sasurau4 committed:
[docs] update segmentedcontrolios deprecation note (#6972)

* [docs] update segmentedcontrolios deprecation note

* [docs] update segmentedcontrolios deprecation note for unversioned

ccheever committed:
[docs] Improve AppAuth docs (#6876)

* [docs] Improve AppAuth docs

- Explain OAuth a little bit more
- Mention that there are specialized modules for Google and Facebook
- Give a complete example as an inline Snack rather than just as
    sourc code

* Update docs/pages/versions/unversioned/sdk/app-auth.md

lg

Co-Authored-By: Evan Bacon <baconbrix@gmail.com>

* Update docs/pages/versions/unversioned/sdk/app-auth.md

lg

Co-Authored-By: Evan Bacon <baconbrix@gmail.com>

* Apply suggestions from code review

Co-Authored-By: Evan Bacon <baconbrix@gmail.com>

* Apply suggestions from code review

Co-Authored-By: Evan Bacon <baconbrix@gmail.com>

Co-authored-by: Evan Bacon <baconbrix@gmail.com>

sjchmiela committed:
[expo-notifications] Add NotificationChannelsManager (#6842)

# Why

`NotificationChannelsManager` singleton module will be the manager of notification channels — modules should proxy any changes to channels through it. (There are no modules allowing to create, modify or delete notification channels at the moment, so the only thing this module does is providing a fallback notification channel — a channel `expo-notifications` creates for notifications not assigned to any other channel. It's a rather common practice — [FCM does it too](https://stackoverflow.com/a/45930522).)

# How

Added a singleton module, which on appropriate platforms (Android && SDK > .O) tries to create a notification channel reactively (only when asked for a fallback channel). Name is fetched from Android resources, so it will should easy to overwrite or localize by developers.

![excalidraw-202012311929-12](https://user-images.githubusercontent.com/1151041/73084749-d28e9980-3ecd-11ea-850d-f76b176c2040.png)

More complete diagram (receiving + handling + building + side effects)

![excalidraw-202012311929-13](https://user-images.githubusercontent.com/1151041/73084918-1e414300-3ece-11ea-8096-f228d6ceb835.png)

# Test Plan

I have confirmed that when a new app is installed and no other action is taken, the settings say that the app hasn't created any channels yet. Once `.getFallbackChannel()` is called, settings say that there exists a single _"Miscellaneous"_ channel.

sjchmiela committed:
[expo-notifications] Add getExpoPushTokenAsync() (#6733)

# Why

We will want to still allow developers to use Expo backend for sending push notifications.

# How

Implemented fetching of the token in JS.

> Note: In Expo client, where we wouldn't want to expose device push token to developers, we will export a native `getExpoPushTokenAsync` method that will implement this behavior.

# Test Plan

Tested on iOS 13, Android 7.1 and 9.

- ✅ added test to `test-suite` that checks whether fetching of the token renders proper, stringy result passes. It is run on iOS and Android (backend not only doesn't support Web push notifications, but it also doesn't allow requests from origins other than `*.expo.io`).

- ✅ notification sent with Expo push notification tool to the Expo token received reaches the device

  To be able to test whether the notification does reach the device I had to manually call `Permissions.askAsync(Permissions.USER_FACING_NOTIFICATIONS)` since it seems that our server can't send silent notifications (the device was receiving them only when sent via device push token).

  Moreover, I added one of the Exponent push tokens to `@exponent` account and assigned the push token to `(@exponent/bare-expo, dev.expo.Payments)` app/experience.

<center><img width="375" alt="obrazek PNG 6" src="https://user-images.githubusercontent.com/1151041/72167596-b2d67c00-33cb-11ea-8416-1c640d83ae77.png"></center>

sjchmiela committed:
[bare-expo] Install pods

mczernek committed:
Update documentation.

mczernek committed:
Add BackgroundFetch and TaskManager to bare-expo app.

mczernek committed:
Update android/expoview/src/main/java/host/exp/exponent/taskManager/ExpoHeadlessAppLoader.java

Co-Authored-By: Tomasz Sapeta <1714764+tsapeta@users.noreply.github.com>

mczernek committed:
Update android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.java

Co-Authored-By: Tomasz Sapeta <1714764+tsapeta@users.noreply.github.com>

mczernek committed:
Update android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.java

Co-Authored-By: Tomasz Sapeta <1714764+tsapeta@users.noreply.github.com>

mczernek committed:
Update android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.java

Co-Authored-By: Tomasz Sapeta <1714764+tsapeta@users.noreply.github.com>

tsapeta commented on packages/@unimodules/react-native-adapter/android/build.gradle:
```diff
@@ -1,4 +1,5 @@
 buildscript {
+  ext.kotlin_version = '1.3.61'
```
Imho we should leverage `ext.kotlinVersion` which can be set by the root project and then use it like this:
```
safeExtGet('kotlinVersion', '1.3.61')
```

mczernek committed:
Apply suggestions from code review

Co-Authored-By: Tomasz Sapeta <1714764+tsapeta@users.noreply.github.com>

mczernek committed:
Throw error if application does not implement ReactApplication.

tsapeta commented on packages/@unimodules/react-native-adapter/android/build.gradle:
```diff
@@ -65,7 +67,9 @@ dependencies {
   unimodule 'unimodules-font-interface'
   unimodule 'unimodules-permissions-interface'
   unimodule 'unimodules-image-loader-interface'
+  unimodule 'expo-app-loader-provider'
```
As you haven't applied this suggestion, I guess this dependency here is required, am I right? If not, then we should remove this - it smells a little bit that we use `expo-*` thing here, in `@unimodules/*` package which should be Expo-agnostic 🤔

mczernek committed:
Move expo-app-loader-provider to unimodules-app-loader

mczernek committed:
Remove JS code from new App Loader unimodule. Remove dependency on old one.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare

mczernek committed:
Revert react-native-unimodules.

mczernek committed:
Update build script for react-native-adapter.

mczernek committed:
Correct expo-notifications build output.

mczernek committed:
Update Podfile.lock.

mczernek committed:
Update bare-expo pods.

mczernek committed:
EXSQlite -> EXSQLite.

tsapeta commented on packages/unimodules-app-loader/android/build.gradle:
```diff
@@ -0,0 +1,71 @@
+
+  defaultConfig {
+    minSdkVersion 21
+    targetSdkVersion 26
```
Use `safeExtGet` in all those sdk versions

tsapeta reviewed (commented):
Hopefully last pass of the review 😄

mczernek committed:
Update build scripts. Revers AppIcon Contents.json.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare

mczernek committed:
Change constant declaration to static ot aviod duplication of symbols.

mczernek committed:
Fix dependencies.

mczernek committed:
Update react-native-unimodules submodule.

tsapeta commented on packages/unimodules-app-loader/android/build.gradle:
```diff
@@ -0,0 +1,76 @@
+  dependencies {
+    classpath 'com.android.tools.build:gradle:3.5.1'
+  }
+}
```
Remove this buildscript block, we no longer declare gradle plugin as a build dependency in our packages

tsapeta reviewed (approved):
Looks good now 👍 just one, small nitpicking comment 👇

mczernek committed:
Remove buildscript section from build.gradle in unimodule-app-loader.

mczernek committed:
Merge branch 'master' into @mczernek/task-manager-bare
